src/hooks/private/useIsomorphicLayoutEffect/useIsomorphicLayoutEffect.ts

```diff
@@ -0,0 +1,4 @@
+import * as React from 'react';
+
+export const useIsomorphicLayoutEffect =
+    typeof window !== 'undefined' ? React.useLayoutEffect : React.useEffect;
```

Comment:
```suggestion
export const useLayoutEffect = typeof window === 'undefined' ? () => {} : React.useLayoutEffect;
```

Using the name `useLayoutEffect` allows to lint the hook's dependencies. There is also no need to call useEffect as it has no effect on the server.